core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -347,6 +347,18 @@ private static boolean skipItem(RexNode expr) {
         .check();
   }
 
+  /**
+   * Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6887">[CALCITE-6887]
+   * In should distinct values with ReduceExpressionRule</a>. */
+  @Test void testReduceExpressionsWithIn() {
+    final String sql = "select deptno, sal "
+        + "from emp "
+        + "where deptno in (1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,3,1) ";
+    sql(sql).withRule(CoreRules.PROJECT_REDUCE_EXPRESSIONS)
```

Comment:
I would like to see a test case where the values are not sorted. `in (2, 1, 2, 3)` should become `in (2, 1, 3)`. This ensures stability.

Reply:
Yes, this is work well because values number in IN lower than 20, would transform to OR with multiple EQUAL and distinct the values. so I did not add case about with test it local. @julianhyde

Reply:
This comment seems important, it's not been incorporated into #4371. I understand you need at least 20 elements to avoid the `OR` expansion, it's OK, just make sure you have a non-sorted example with at least 20 elements as Julian suggests.

Reply:
Thanks for reminder, and I would add this to PR. @asolimando

Reply:
@asolimando @julianhyde I had added the case with non-sorted and less then 20 values in commit: https://github.com/apache/calcite/pull/4371/commits/c4273c449855a8dc8caa48c0b3f49d27a782495e